ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/controlModules/foot/MoveViaWaypointsState.java

```diff
@@ -214,6 +214,8 @@ public FeedbackControlCommand<?> createFeedbackControlTemplate()
    @Override
    public void onExit(double timeInState)
    {
+      controlManager.getTaskspaceControlState().onExit(timeInState);
+      controlManager.getJointspaceControlState().onExit(timeInState);
```

Comment:
@SylvainBertrand if we don't call this, the desired state doesn't necessarily get reset. Then, if you put the foot down, awlk forward, and lift it again, it starts with the first waypoint as the old terminal position. I set up a better test in EndToEndFootTrajectoryMessageTest that showed this.